aiohttp/http_parser.py

```diff
@@ -668,12 +668,26 @@ def feed_data(self,
                 # we should get another \r\n otherwise
                 # trailers needs to be skiped until \r\n\r\n
                 if self._chunk == ChunkState.PARSE_MAYBE_TRAILERS:
-                    if chunk[:2] == SEP:
-                        # end of stream
-                        self.payload.feed_eof()
-                        return True, chunk[2:]
+                    if len(chunk) >= 2:
```

Comment:
```python
head = chunk[:2]
if not head:
    return False, b''
if head == SEP[:1]:
    self._chunk_tail = head
    return False, b''
if head == SEP:
    # end of stream
    self.payload.feed_eof()
    return True, chunk[2:]
self._chunk = ChunkState.PARSE_TRAILERS
```

Reply:
Hello @socketpair
 
I agree with you. So I changed the code simpler as you suggested.
But I moved the `if head == SEP' at the top because it is more likely to happen.

Thanks,